File: app/javascript/graphql/mutations/signup.js
import gql from 'graphql-tag';
export const SIGNUP_USER = gql`
mutation($email: String!, $password: String!,$firstName: String!,$lastName: String!) {
  signUp(
    email: $email,
    password: $password,
    firstName: $firstName,
    lastName: $lastName,
  ) {
    email
    accessToken
    firstName
    lastName
  }
}
`;

File: app/graphql/mutations/signup.rb
# frozen_string_literal: true

module Mutations
  class Signup < GraphQL::Schema::Mutation
    null true

    description 'Sign up a user'

    argument :email, String, required: true
    argument :password, String, required: true
    argument :firstName, String, required: true
    argument :lastName, String, required: true

    field :email, String, null: false
    field :access_token, String, null: false
    field :firstName, String, null: false
    field :lastName, String, null: false

    def resolve(**args)
      user = User.create(email: args[:email], password: args[:password], first_name: args[:firstName], last_name: args[:lastName])
      access_token = user.access_tokens.create
      {
        firstName: user.first_name,
        lastName: user.last_name,
        email: user.email,
        access_token: access_token.token
      }
    end
  end
end
